cpp/ps0006/problem 7.cpp

```diff
@@ -18,7 +18,7 @@ int main(void)
     counter(10, 10, 0); // 10
     std::cout << "===============" << std::endl;
```

Comment:
- [x] `start=0`, `step=1` as default
- [x] up counter works properly when `step` is positive
- [x] down counter works properly when `step` is negative